apps/backend-sls/test/integration.test.ts

```diff
@@ -1,13 +1,241 @@
+import { PrismaClient } from '@klicker-uzh/prisma'
+import Redis from 'ioredis'
 import request from 'supertest'
+import prepareApp from '../GraphQL/app'
 
-import app from '../src/app'
+// TODO: switch to ioredis-mock
+// jest.mock('ioredis', () => Redis)
 
-describe('GraphQL API', () => {
-  it('works', async () => {
+process.env.NODE_ENV = 'development'
+
+describe('API', () => {
+  let prisma: PrismaClient
+  let redisCache: Redis
+  let redisExec: Redis
+  let app: Express.Application
+  let userCookie: string = ''
+  let session: any
+
+  beforeAll(() => {
+    prisma = new PrismaClient()
+    redisCache = new Redis({
+      family: 4,
+      host: process.env.REDIS_CACHE_HOST ?? 'localhost',
+      password: process.env.REDIS_CACHE_PASS ?? '',
+      port: Number(process.env.REDIS_CACHE_PORT) ?? 6379,
+      tls: process.env.REDIS_CACHE_TLS ? {} : undefined,
+    })
+    redisExec = new Redis({
+      family: 4,
+      host: process.env.REDIS_HOST ?? 'localhost',
+      password: process.env.REDIS_PASS ?? '',
+      port: Number(process.env.REDIS_PORT) ?? 6380,
+      tls: process.env.REDIS_TLS ? {} : undefined,
+    })
+  })
+
+  afterAll(async () => {
+    await redisCache.quit()
+    await redisExec.quit()
+  })
+
+  beforeEach(() => {
+    app = prepareApp({ prisma, redisCache, redisExec })
+  })
+
+  it('allows logging in a user', async () => {
+    const response = await request(app)
+      .post('/api/graphql')
+      .send({
+        query: `
+        mutation {
+          loginUser(email: "roland.schlaefli@bf.uzh.ch", password: "abcd")
+        }
+      `,
+      })
+
+    expect(response.headers['set-cookie']).toBeDefined()
+    userCookie = response.headers['set-cookie'][0]
+
+    expect(response.body).toMatchInlineSnapshot(`
+      Object {
+        "data": Object {
+          "loginUser": "6f45065c-447f-4259-818c-c6f6b477eb48",
+        },
+        "extensions": Object {
```

Comment:
This is provided by the response cache plugin, which will dynamically store query and mutation results in memory, and invalidate them with heuristics. Works quite well, but keep in mind when developing (e.g., check that errors are not caused due to caching).